### PR TITLE
`--only` flag for the build-latest-rc command

### PR DESCRIPTION
**Pull request by iowillhoit**
### What does this PR do?
Allows the build-latest-rc script to *only* update the packages that are passed to the `--only` flag. This is especially useful for `patch` releases

### What issues does this PR fix or reference?
[@W-11462358@](https://gus.my.salesforce.com/apex/ADM_WorkLocator?bugorworknumber=W-11462358)

### CODE REVIEW

**Review by RodEsp — CHANGES_REQUESTED** [review]
Looks good overall, just some minor comments.

File: src/package.ts [hunk 1]
```diff
@@ -151,25 +160,89 @@ export class Package extends AsyncOptionalCreatable {
...
+          // package name was passed, so we can use key and value directly
+          return {
+            dependencyName: key,
+            packageName: key,
+            alias: null,
+            currentVersion: value,
+          };
+        }
+      }
+      if (value.includes(`npm:${name}`)) {
```
[the remark this answers is not in the document]
Comment (iowillhoit) [reply src/package.ts:208]: Yea, I don't think this would matter but your suggestion more explicit to someone reading the code. Thanks!

**Review by RodEsp — APPROVED** [review]

**Review by mshanemc — APPROVED** [review]
cool stuff

**Comment by mshanemc** [comment]
QA notes:

❌ `sfdx cli:latestrc:build --build-only --only @salesforce/source-deploy-retrieve`
if we want to bump a library that's in resolutions but not in `dependencies`, can the command do that?  between core and sdr, I figure these might be be as common as plugin-bumps.

👎🏻  `sfdx cli:latestrc:build --build-only --only @salesforce/plugin-auth` (pick a plugin where there are no changes)
maybe tell people that there are no changes in what they specified?
bonus points: check that first and exit rather than making branches, changing versions, opening PRs.

--only salesforce-alm
✅ bumps toolbelt as expected
❓  but there aren't any other changes in other plugins/resolutions that can verify that only toolbelt will change 

🤔 probably and unrelated note: snapshot is formatted completely differently
Screen Shot 2022-07-25 at 10 20 04 AM

File: src/package.ts [hunk 2]
```diff
@@ -151,25 +160,93 @@ export class Package extends AsyncOptionalCreatable {
...
+      // regex for npm package with optional namespace and version
+      // https://regex101.com/r/HmIu3N/1
+      const npmPackageRegex = /^((?:@[^/]+\/)?[^@/]+)(?:@([^@/]+))?$/;
+      const [, name, version] = npmPackageRegex.exec(dep);
+
+      // find dependency in package.json (could be an npm alias)
+      const depInfo = this.getDependencyInfo(name);
+
+      // If a version is not provided, we'll look up the "latest" version
+      depInfo.finalVersion = version ?? this.getDistTags(depInfo.packageName).latest;
```
[the remark this answers is not in the document]
Comment (iowillhoit) [reply src/package.ts:235]: Just to make sure I understand what you are suggesting for the latter, change this to (pseudo): 
`finalVersion = version ?? tags['latest-rc'] ?? tags['latest']`

**Review by mshanemc — APPROVED** [review]

**Comment by mshanemc** [comment]
qa:
✅ only on a resolutions-only will result in good message (No version changes made), as does a non-changed plugin
✅ bumps one plugin when several would normally be bumped


not tested: bumping a resolution that has changed (there are none)

**Pull request merged**